xarray/tests/test_computation.py

```diff
@@ -670,7 +685,11 @@ def check(x, y):
     check(data_array, 0 * data_array)
     check(data_array, 0 * data_array[0])
     check(data_array[:, 0], 0 * data_array[0])
-    check(data_array, 0 * data_array.compute())
+    # todo: this raises ValueError since chunks do not match inside apply_gufunc due to
+    #  computing, could be fixed by setting `allow_rechunk=True` in the call to
+    #  apply_gufunc (or rechunking before feeding to apply_ufunc)
+    with raises_regex(ValueError, "with different chunksize present"):
```

Comment:
It seems it does allow it if `allow_rechunk=True`.

Reply:
Hmm, I don't think I fully understood this before.

My original thinking was that passing a NumPy array into `apply_gufunc` shouldn't really require `allow_rechunk=True`, because even though the NumPy array has different chunks, there is a difference between "no chunks" and "different chunks".

It seems like it should be OK for `apply_gufunc` to handle "no chunks"? In that case, the NumPy array could be given the same chunks as the other dask arrays.

Not a huge deal to include this change (there is an easy work around), but I could be worth fixing upstream in dask.